lib/query_diet/widget.rb

```diff
@@ -31,9 +31,19 @@ def css
         EOF
       end
 
+      def js(nonce_attribute)
+        <<-EOF
+        <script type="text/javascript"#{nonce_attribute}>
+          document.getElementById("query_diet").addEventListener("click", function() {
+            this.parentNode.removeChild(this);
+          });
+        </script>
+        EOF
+      end
+
       def html(options)
         <<-EOF
-        <div id="query_diet" class="#{QueryDiet::Logger.bad?(options) ? 'bad' : 'good' }" onclick="this.parentNode.removeChild(this);">
+        <div id="query_diet" class="#{QueryDiet::Logger.bad?(options) ? 'bad' : 'good' }">
```

Comment:
I think you need two different variants here:

* If `nonce: true` remove the line like you did
* If `nonce: false` keep it like it is (so it still works for projects without CSP or `:unsafe_inline`

Example:

```
<div id="query_diet" class="#{QueryDiet::Logger.bad?(options) ? 'bad' : 'good' }" #{options.fetch(:nonce) ? '' : 'onclick="this.parentNode.removeChild(this);"'}>
```

```

```